### PR TITLE
Project hello-rails-react: Set-up

### PR DESCRIPTION
**Pull request by sarazaToussaint**
### In this Pull request I:

- Create the App component with react-router and render it in a static view.
- Create the Greeting component that will display a greeting. Set it up as a route in the App component.
- Create a store, an action, and a reducer that will connect to an API endpoint to get the random greeting.
- Display the random greeting in the Greeting component.

### CODE REVIEW

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    });
+  }
+
+  useEffect(() => {
+    fetchMessage();
+  }, []);
+
+  return (
+    <div>
+      <h1>{message}</h1>
```
Comment (julie-ify) [inline app/javascript/components/Greeting.js:22]: - [x] Despite installing all dependencies and setting up the app correctly, nothing is displayed when I visit your app. 

  lod

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @sarazaToussaint,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/Greeting.js (hunk 1 above)
Comment (sarazaToussaint) [reply app/javascript/components/Greeting.js:22]: Hi @julie-ify I solved the issue, and now my app is working perfectly. anyway thank you for your review

**Review by ShahierNashaat — APPROVED** [review]
Hi @sarazaToussaint, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sarazaToussaint** [comment]
Hi @ShahierNashaat, thank you for reviewing and approving my PR.

**Pull request merged**